src/Adapter/Presenter/Cart/CartPresenter.php

```diff
@@ -475,6 +473,41 @@ function ($voucher) {
                 '',
         );
     }
+    
+    private function getShippingDisplayValue($cart, $shippingCost) 
+    {        
+        $hasFreeCarrierFirst = 0;
+        $default_country = null;
+        
+        if (isset(Context::getContext()->cookie->id_country)) {
+            $default_country = new Country(Context::getContext()->cookie->id_country);
+        }
+
+        $delivery_option_list = $cart->getDeliveryOptionList($default_country);
+
+        if (isset($delivery_option_list) && count($delivery_option_list) > 0) {
+            foreach ($delivery_option_list as $option) {
+                foreach ($option as $currentCarrier) {
+                    if ($currentCarrier['position'] == 0 && (isset($currentCarrier['is_free']) && $currentCarrier['is_free'] > 0)) {
+                        $hasFreeCarrierFirst = 1;
+                        break;
+                    }
+                }
+            }
+        }
+
+        $shippingDisplayValue = '';
+
+        if ($shippingCost != 0) {
+            $shippingDisplayValue = $this->priceFormatter->format($shippingCost);
+        } else if ($hasFreeCarrierFirst == 0) {
+            $shippingDisplayValue = $this->translator->trans('Not Calculated', array(), 'Shop.Theme.Checkout');
```

Comment:
Hey, thank you, wouldn't it be easier to remove this line when shipping costs are not determined yet and display it only when free shipping is defined in parameters? If not, I would suggest _Calculated later_ or _Calculated at checkout_. Ping me if you all have a better idea!

Reply:
@LouiseBonnard Yes"Calculated later" seems better. If anyone else has a better idea let's hear it.

Reply:
I would vote for the first suggestion from @LouiseBonnard, removing this line when not calculated...

Reply:
@marionf @rblaurin @LouiseBonnard 

Yes we can hide it. I think we need to change the theme in two places (if more please let me know).

First we send an empty string instead of "Not Calculated".

For modal cart we must change this file:
themes\classic\modules\ps_shoppingcart\modal.tpl on line 60 and surround it with an if statement to check "$cart.subtotals.shipping.value"

For checkout page this file: themes\classic\templates\checkout\_partials\cart-summary.tpl on line 56 we change "if $subtotal" with "if $subtotal.value"

The file themes\classic\templates\checkout\_partials\cart-detailed-totals.tpl is already doing this on line 30 so we don't need to change anything.

I tested this on 1.7.5.2 and works fine. Are there other places we need to change?